src/main/kotlin/org/digma/intellij/plugin/ui/list/insights/InsightsListCellRenderer.kt

```diff
@@ -102,7 +102,13 @@ class InsightsListCellRenderer : AbstractPanelListCellRenderer() {
         val endpoint =  routeInfo.shortName
         if(routeInfo.schema == HTTP_SCHEMA){
             val split =endpoint.split(' ')
-            return GroupViewModel(asHtml("${spanBold(split[0].uppercase())} ${span("${split[1].uppercase()} ${split[2]}")}"),  Laf.Icons.Insight.INTERFACE)
+            val lastElementIndex = split.size - 1
+            val httpMethodWithRoute = span("${split[lastElementIndex - 1].uppercase()} ${split[lastElementIndex]}")
+            return if (split.size < 3) {
+                GroupViewModel(asHtml("${spanBold("HTTP")} $httpMethodWithRoute"),  Laf.Icons.Insight.INTERFACE)
+            } else {
+                GroupViewModel(asHtml("${spanBold(split[lastElementIndex - 2].uppercase())} $httpMethodWithRoute"), Laf.Icons.Insight.INTERFACE)
```

Comment:
There's some repetition here. Can you extract the schema name to a variable and then have one line calling `GroupViewModel`?

Reply:
fixed